### PR TITLE
feat(macros.js): add a toJSON method to vtkObject publicAPI

### PR DESCRIPTION
**Pull request by Julesdevops**
The toJSON method simply returns the result of publicAPI.getState()
It allow user to call JSON.stringify directly on a vtkObject instead of doing JSON.stringify(vtkObject.getState())

### PR and Code Checklist
- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Context
To serialize an object to JSON you have to use JSON.stringify(vtkObj.getState())

### Changes
Now we can use JSON.stringify(vtkObj) and we will get the same result.
- [x] Documentation and TypeScript definitions were updated to match those changes

### Testing
- [ ] This change adds or fixes unit tests
- [ ] All tests complete without errors on the following environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**:

### CODE REVIEW

File: Sources/macros.js [hunk 1]
```diff
@@ -391,6 +391,12 @@ export function obj(publicAPI = {}, model = {}) {
     publicAPI.modified();
   };
 
+  // This function will get called when one invoke JSON.stringify(vtkObject)
+  // JSON.stringify will only stringify the return value of this function
+  publicAPI.toJSON = function vtkObjToJSON() {
```
Comment (jourdain) [inline Sources/macros.js:396]: why not write `publicAPI.toJSON = publicAPI.getState`

Comment (finetjul) [reply Sources/macros.js:396]: if `getState()` get overriden in subclass

Comment (jourdain) [reply Sources/macros.js:396]: Make sense...

**Comment by floryst** [comment]
Tests are hanging for some reason. Can you isolate the test and verify that it runs locally?

**Comment by Julesdevops** [comment]
> Tests are hanging for some reason. Can you isolate the test and verify that it runs locally?

Tests are running fine locally ; Do you want me to create a whole new test/test file for this ?

**Comment by finetjul** [comment]
Failing tests seem to be due to WidgetManagerTest

**Comment by finetjul** [comment]
@jourdain WDYT about the solution to fix infinitely recursive `getState()` ? I'm not a big fan (especially for `_parentProp` in `vtkProp`), but if that's fine with you, I'm good with it. Another option would be to use "unique references" to already serialized objects to avoid infinite recursions...

File: Sources/Filters/General/AppendPolyData/test/testAppendPolyData.js [hunk 2]
```diff
@@ -66,17 +66,16 @@ test('Test vtkAppendPolyData execution', (t) => {
 });
 
 test('Test addInputData edge case', (t) => {
-  const gc = testUtils.createGarbageCollector(t);
-  const appender = gc.registerResource(vtkAppendPolyData.newInstance());
```
Comment (jourdain) [inline Sources/Filters/General/AppendPolyData/test/testAppendPolyData.js:70]: Why do you remove the GC in that test?

Comment (finetjul) [reply Sources/Filters/General/AppendPolyData/test/testAppendPolyData.js:70]: It is not an asynchronous test, hence there is no need for GC. Did I miss something ?

**Comment by finetjul** [comment]
Thanks for your review.

Ok for `macro.moveToPrivate()`. We could rename it to `macro.private` (to be more consistant with `macro.get` and `macro.set` ?)

We could also handle in `moveToPrivate()` the "_" prefix renaming of set/get accessors ? Or is it fine to leave the renaming to `capitalize()`? My motivation to handle the set/get renaming in `moveToPrivate()` is that the usefulness of `moveToPrivate()` is not very obvious because it will work fine without it for most of the times. I'm afraid developers will forget it. They won't forget it if they need accessors.

Btw, `capitalize()` works fine (i.e. `_interactor` -> `Interactor` and `interactor` -> `Interactor`), but `uncapitalize()` is more limited (i.e. `Interactor` -> `interactor`, it will never do `Interactor` -> `_interactor`).

**Comment by jourdain** [comment]
I was limiting the use of `moveToPrivate` to only rename key in dict if available. 

Not all private var should have set/get. But extending set/get macro to better handle them make sense.

File: Sources/Filters/General/AppendPolyData/test/testAppendPolyData.js (hunk 2 above)
Comment (finetjul) [reply Sources/Filters/General/AppendPolyData/test/testAppendPolyData.js:70]: Are you saying ALL tests must use testUtils GarbageCollector ?

Comment (jourdain) [reply Sources/Filters/General/AppendPolyData/test/testAppendPolyData.js:70]: Yes otherwise, the canvas stay around (as nobody delete them) and we run out of available one.  The gc in test help to clear things between tests.

Comment (jourdain) [reply Sources/Filters/General/AppendPolyData/test/testAppendPolyData.js:70]: I mean the rendering ones. Having leaks in test with objects is not a big deal, but gl canvas are limited to 16 or something like that.
